zpay32/invoice.go

```diff
@@ -287,21 +283,36 @@ func Decode(invoice string) (*Invoice, error) {
 	// address. This will also determine which network this invoice is
 	// meant for.
 	var net *chaincfg.Params
-	if strings.HasPrefix(hrp[2:], chaincfg.MainNetParams.Bech32HRPSegwit) {
+	switch {
+	case strings.HasPrefix(hrp[2:], chaincfg.MainNetParams.Bech32HRPSegwit):
 		net = &chaincfg.MainNetParams
-	} else if strings.HasPrefix(hrp[2:], chaincfg.TestNet3Params.Bech32HRPSegwit) {
+	case strings.HasPrefix(hrp[2:], chaincfg.TestNet3Params.Bech32HRPSegwit):
 		net = &chaincfg.TestNet3Params
-	} else if strings.HasPrefix(hrp[2:], chaincfg.SimNetParams.Bech32HRPSegwit) {
+	case strings.HasPrefix(hrp[2:], chaincfg.SimNetParams.Bech32HRPSegwit):
 		net = &chaincfg.SimNetParams
-	} else {
+	case strings.HasPrefix(hrp[2:], chaincfg.RegressionNetParams.Bech32HRPSegwit):
```

Comment:
The `Bech32HRPSegwit` prefix of [`TestNet3Params`](https://github.com/Roasbeef/btcd/blob/9978b939c33973be19b932fa7b936079bb7ba38d/chaincfg/params.go#L473) and [`RegressionNetParams`](https://github.com/Roasbeef/btcd/blob/9978b939c33973be19b932fa7b936079bb7ba38d/chaincfg/params.go#L382) is the same: `tb`, so I'd be careful of doing the check this way. I simply opted not to check for regtest.